### PR TITLE
Show total number of sponsors in general information

### PR DESCRIPTION
**Pull request by chrismaree**
This small PR adds the total number of sponsors within a given EMP to the general Info tab.
image

### CODE REVIEW

File: features/contract-state/DisputeParams.tsx [hunk 1]
```diff
@@ -23,7 +30,11 @@ const DisputeParams = () => {
     disputeBondPct,
     disputerDisputeRewardPct,
     sponsorDisputeRewardPct,
+    withdrawalLiveness,
   } = empState;
+  const withdrawalLivenessInMinutes = (Number(withdrawalLiveness) / 60).toFixed(
```
Comment (nicholaspai) [inline features/contract-state/DisputeParams.tsx:35]: This withdrawal liveness is already shown in general info?

File: features/contract-state/GeneralInfo.tsx [hunk 2]
```diff
@@ -40,11 +41,16 @@ const GeneralInfo = () => {
...
     collateralRequirement: collReq,
     minSponsorTokens,
-    withdrawalLiveness,
   } = empState;
   const { symbol: tokenSymbol } = Token.useContainer();
+
+  const [sponsorCount, setSponsorCount] = useState<string | null>(null);
   const defaultMissingDataDisplay = "N/A";
 
+  useEffect(() => {
```
Comment (nicholaspai) [inline features/contract-state/GeneralInfo.tsx:50]: `activeSponsors` is already updated within an `useEffect` hook in the EMPSponsors container, so why not just add `activeSponsors !== null` to the big `if` statement, and then just create a variable `const sponsorCount = Object.keys(activeSponsors).length`? I don't think you need to create the state variable as you did

File: features/contract-state/GeneralInfo.tsx [hunk 3]
```diff
@@ -60,11 +66,9 @@ const GeneralInfo = () => {
     ).toLocaleString("en-GB", { timeZone: "UTC" });
     const prettyLatestPrice = Number(latestPrice).toFixed(4);
     const pricedGcr = (gcr / latestPrice).toFixed(4);
-    const withdrawalLivenessInMinutes = (
-      Number(withdrawalLiveness) / 60
-    ).toFixed(2);
+
     const priceIdUtf8 = parseBytes32String(priceId);
-    const collReqPct = (parseFloat(fromWei(collReq)) * 100).toString();
+    const collReqPct = parseFloat(fromWei(collReq)).toString();
```
Comment (nicholaspai) [inline features/contract-state/GeneralInfo.tsx:71]: +1 as its more consistent with how we display CR in other files

File: features/contract-state/DisputeParams.tsx (hunk 1 above)
Comment (chrismaree) [reply features/contract-state/DisputeParams.tsx:35]: nope. I removed it from general and put it in the dispute section. see the screenshot for how it looks now.

File: features/contract-state/GeneralInfo.tsx (hunk 2 above)
Comment (chrismaree) [reply features/contract-state/GeneralInfo.tsx]: yup, that's a cleaner way of doing it. I didn't know I could access the `activeSponsors` like that from the EMPSponsors container. I have refactored, PTAL.

**Comment by chrismaree** [comment]
I have implemented your nits, please take a look @nicholaspai

File: features/contract-state/GeneralInfo.tsx [hunk 4]
```diff
@@ -1,37 +1,38 @@
 import styled from "styled-components";
 import { utils } from "ethers";
 import { Typography, Box, Tooltip } from "@material-ui/core";
+import { useState, useEffect } from "react";
```
Comment (nicholaspai) [inline features/contract-state/GeneralInfo.tsx:4]: You can remove `useState` and `useEffect` now right?

Comment (chrismaree) [reply features/contract-state/GeneralInfo.tsx]: ye, you right! have removed this & pushed up for the merge 🚀

**Pull request merged**